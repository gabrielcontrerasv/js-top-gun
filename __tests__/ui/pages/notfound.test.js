import { fireEvent, render, screen } from "@testing-library/react";
import NotFoundPage from "@/pages/404/index";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
    };
  },
}));

const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("NotFoundPage", () => {
  test("Matches DOM Snapshot", () => {
    const wrapper = render(<NotFoundPage />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Should display the 404 Not found Page correctly", () => {
    render(<NotFoundPage />);
    useRouter.mockImplementation(() => ({
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
    }));

    const goHomeBtn = screen.getByRole("button", {
      name: /GO HOME/,
    });

    //fireEvent.click(goHomeBtn);
    

    expect(goHomeBtn).toBeInTheDocument();
  });
});
