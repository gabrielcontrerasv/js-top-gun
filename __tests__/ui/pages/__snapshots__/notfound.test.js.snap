// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Matches DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-start-1 col-end-3 flex justify-center items-center row-span-2 w-full relative"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
        <div
          class="flex flex-col justify-between items-center h-[90%] mt-10 z-10 "
        >
          <div>
            <p
              class="text-[7em] sm:text-[15rem] text-center text-[#565656] font-semibold"
            >
              404
            </p>
          </div>
          <div
            class="flex flex-col justify-center items-center gap-2"
          >
            <h1
              class="text-3xl md:text-4xl text-center font-bold text-[#565656]"
            >
              Uh oh we have a problem
               
            </h1>
            <button
              class="bg-[#154D4D] text-white py-3 px-6 rounded-md tracking-widest hover:bg-[#217a7a] w-[16rem] "
            >
              GO HOME
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-start-1 col-end-3 flex justify-center items-center row-span-2 w-full relative"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
        <noscript />
      </span>
      <div
        class="flex flex-col justify-between items-center h-[90%] mt-10 z-10 "
      >
        <div>
          <p
            class="text-[7em] sm:text-[15rem] text-center text-[#565656] font-semibold"
          >
            404
          </p>
        </div>
        <div
          class="flex flex-col justify-center items-center gap-2"
        >
          <h1
            class="text-3xl md:text-4xl text-center font-bold text-[#565656]"
          >
            Uh oh we have a problem
             
          </h1>
          <button
            class="bg-[#154D4D] text-white py-3 px-6 rounded-md tracking-widest hover:bg-[#217a7a] w-[16rem] "
          >
            GO HOME
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
